src/NodeCompiler/CompileNodeToValue.php

```diff
@@ -47,6 +47,10 @@ public function __invoke(Node $node, CompilerContext $context)
             return $this->compileBinaryOperator($node, $context);
         }
 
+        if ($node instanceof Node\Scalar\MagicConst\Dir) {
+            return '__DIR__';
```

Comment:
I see. Test case fixed and now it fails.

But I'm not sure how to get `__DIR__` value from `$context` or `$node`.